internal/services/machinelearning/machine_learning_workspace_resource.go

```diff
@@ -123,6 +123,19 @@ func resourceMachineLearningWorkspace() *pluginsdk.Resource {
 				DiffSuppressFunc: suppress.CaseDifference,
 			},
 
+			"public_network_access": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+
+			"image_build_compute": {
```

Comment:
this makes no sense, should it instead be
```suggestion
			"compute_image_name": {
```

Reply:
@katbyte We require this property to be named "image_build_compute". This attribute does not set the compute image. Azure Machine Learning can build docker images on behalf of the customer. By default we use Azure Container Registry task compute for that. In case of network isolation this is not possible, and customers are required to provide the name of the Azure ML compute cluster to use for image building.

Reply:
So this is the name of the compute cluster to use for image building?

Reply:
Yes that is correct

Reply:
then we would probably want to call it image_build_compute_cluster_name, to be explicit about what it is for/type of value is expected to hint at the resource type & property expected. and update the docs

Reply:
@denniseik  which name should we use? `image_build_compute_cluster_name` or `image_build_compute`?

Reply:
Please use image_build_compute for consistency across docs and resource providers. Thank you!

Reply:
@katbyte Clarification from @denniseik：the property we are talking is in the document : [Secure an Azure Machine Learning workspace with virtual networks - Azure Machine Learning | Microsoft Docs](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-secure-workspace-vnet#enable-azure-container-registry-acr) and [azureml.core.Environment class - Azure Machine Learning Python | Microsoft Docs](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.environment(class)?view=azure-ml-py#build-workspace--image-build-compute-none-) 
For example:
Currently only CPU-type compute clusters are supported, but we might be supporting other types in the future or rely back on ACR tasks when they provide full network isolation. `image_build_compute` has always been a bit of a workaround because we cannot use ACR tasks as compute when the workspace is private link enabled.

Reply:
what does the portal use?

Reply:
We don't expose this property in the Azure portal deployment experience. It's typically updated post-deployment incrementally. In ARM/CLI/SDK its called like the above.